beacon_node/beacon_chain/src/test_utils.rs

```diff
@@ -888,29 +890,35 @@ where
                 randao_reveal,
                 Some(graffiti),
                 ProduceBlockVerification::VerifyRandao,
+                BlockProductionVersion::FullV2,
             )
             .await
-            .unwrap();
-
-        let signed_block = block.sign(
-            &self.validator_keypairs[proposer_index].sk,
-            &state.fork(),
-            state.genesis_validators_root(),
-            &self.spec,
-        );
+            .unwrap()
+        {
+            let signed_block = block_response.block.sign(
+                &self.validator_keypairs[proposer_index].sk,
+                &block_response.state.fork(),
+                block_response.state.genesis_validators_root(),
+                &self.spec,
+            );
 
-        let block_contents: SignedBlockContentsTuple<E, FullPayload<E>> = match &signed_block {
-            SignedBeaconBlock::Base(_)
-            | SignedBeaconBlock::Altair(_)
-            | SignedBeaconBlock::Merge(_)
-            | SignedBeaconBlock::Capella(_) => (signed_block, None),
-            SignedBeaconBlock::Deneb(_) => (
-                signed_block,
-                maybe_blob_sidecars.map(|blobs| self.sign_blobs(blobs, &state, proposer_index)),
-            ),
-        };
+            let block_contents: SignedBlockContentsTuple<E, FullPayload<E>> = match &signed_block {
+                SignedBeaconBlock::Base(_)
+                | SignedBeaconBlock::Altair(_)
+                | SignedBeaconBlock::Merge(_)
+                | SignedBeaconBlock::Capella(_) => (signed_block, None),
+                SignedBeaconBlock::Deneb(_) => (
+                    signed_block,
+                    block_response
+                        .maybe_side_car
+                        .map(|blobs| self.sign_blobs(blobs, &block_response.state, proposer_index)),
+                ),
+            };
 
-        (block_contents, state)
+            (block_contents, block_response.state)
+        } else {
+            panic!("Should always be a full payload response")
```

Comment:
Yes this can definitely be simplified. Unfortunately clippy doesn't catch this flavor of `if let else` yet. There's an open issue in the clippy repo here: https://github.com/rust-lang/rust-clippy/issues/9807. I'll keep my eye on it as I think this would be another nice linting rule to add to lighthouse in the future. Might even take a stab at implementing the linting rule if I have some time